src/hotspot/cpu/riscv/gc/z/zBarrierSetAssembler_riscv.cpp

```diff
@@ -724,8 +724,8 @@ void ZBarrierSetAssembler::generate_c2_load_barrier_stub(MacroAssembler* masm, Z
   {
     SaveLiveRegisters save_live_registers(masm, stub);
     ZSetupArguments setup_arguments(masm, stub);
-    __ mv(t0, stub->slow_path());
-    __ jalr(t0);
+    __ mv(t1, stub->slow_path());
+    __ jalr(t1);
   }
```

Comment:
Bad news is that I do notice problems with use of t1 register for C2 rflags. Similar issue already exists for JDK head.
See this PR for more details: https://github.com/openjdk/jdk/pull/21485. I think we should get this resolved before this one.

Reply:
Yes, thanks!